enhancements/olm/simplify-olm-apis.md

```diff
@@ -0,0 +1,366 @@
+- they also require a `ClusterRole` to configure any access
+
+Define a label convention for selecting resources that compose an operator
+
+- `operators.coreos.com/v2/operators/<name>: ""`; where `name` is `metadata.name`
+- using a unique label key helps avoid collisions
+- using a deterministic label key lets users know the exact label used in advance
+
+Define, as part of the `Operator` resource, a `status.resources` field that surfaces:
+
+- `matchLabels`: label selector used to list operator components
+- `components`: references to resources selected by `matchLabels`
+
+Add OLM controller logic such that, for each `Operator` resource:
+
+- `matchLabels` is resolved from the label convention and stored in the `status.resources.matchLabels` field
+
+```yaml
+status:
+  resources:
+    matchLabels:
+      operators.coreos.com/v2/operators/plumbus: ""
+```
+
+Add OLM controller logic such that, for each `Operator` resource:
+
+- cluster-scoped resources are selected using `status.resources.matchLabels`
+- namespace-scoped resources are selected __across all namespaces__ using `status.resources.matchLabels`
+- the union of both selections is stored in `status.resources.components`
+- kinds selected in both cases may be predefined or queried from discovery
+
+```yaml
+status:
+  resources:
+    matchLabels:
+      operators.coreos.com/v2/operator-name: plumbus
+    components:
+    - kind: Subscription
+      namespace: my-ns
+      name: plumbus
+      uid: d655a13e-d06a-11e9-821f-9a2e3b9d8156
+      apiVersion: operators.coreos.com/v1alpha1
+      resourceVersion: 109719
+    - kind: ClusterServiceVersion
+      namespace: my-ns
+      name: plumbus.v2.0.0-alpha
+      uid: d70a53b5-d06a-11e9-821f-9a2e3b9d8156
+      apiVersion: operators.coreos.com/v1alpha1
+      resourceVersion: 109811
+    # ...
+```
+
+Add OLM controller logic, such that for each `Operator` resource
+
+- OLM resources matching `status.resources.matchLabels` have the matching labels projected onto all related generated resources
+  - from `Subscriptions` onto resolved `Subscriptions` and `InstallPlans`
+  - from `InstallPlans` onto `ClusterServiceVersions`, `CustomResourceDefinitions`, RBAC, etc.
+  - from `ClusterServiceVersions` onto `Deployments` and `APIServices`
+
+Define, as part of the `Operator` resource, a `status.metadata` field that
+
+- contains the `displayName`, `description`, and `version` of an operator
+
+```yaml
+status:
+  metadata:
+    displayName: Plumbus
+    description: Welcome to the exciting world of Plumbus ownership! A Plumbus will aid many things in life, making life easier. With proper maintenance, handling, storage, and urging, Plumbus will provide you with a lifetime of better living and happiness.
+    version: 2.0.0-alpha
+```
+
+Define, as part of the `Operator` resource, a `status.apis` field that
+
+- contains the `required` and `provided` APIs of an operator
+
+```yaml
+status:
+  apis:
+    provides:
+    - group: how.theydoit.com
```

Comment:
This doesn't have to be here, but if we want the UI to switch over to display `Operator`s instead of `ClusterServiceVersions`, we need somewhere for the descriptor data to live.

Reply:
I ended up nesting `apis` under `status.metadata`. We can also put the descriptors in there if we decide to phase out CSVs.